delivery-receipts-api/src/main/java/uk/gov/di/authentication/deliveryreceiptsapi/lambda/NotifyCallbackHandler.java

```diff
@@ -0,0 +1,117 @@
+        this.configurationService = configurationService;
+    }
+
+    public NotifyCallbackHandler(ConfigurationService configurationService) {
+        this.cloudwatchMetricsService = new CloudwatchMetricsService(configurationService);
+        this.configurationService = configurationService;
+    }
+
+    public NotifyCallbackHandler() {
+        this(ConfigurationService.getInstance());
+    }
+
+    @Override
+    public APIGatewayProxyResponseEvent handleRequest(
+            APIGatewayProxyRequestEvent input, Context context) {
+        LOG.info("Received request");
+        validateBearerToken(input.getHeaders());
+        NotifyDeliveryReceipt deliveryReceipt;
+        try {
+            deliveryReceipt =
+                    new ObjectMapper().readValue(input.getBody(), NotifyDeliveryReceipt.class);
+            if (deliveryReceipt.getNotificationType().equals("sms")) {
+                var countryCode = getCountryCodeFromNumber(deliveryReceipt.getTo());
+                var deliveryStatus = getDeliveryStatus(deliveryReceipt.getStatus());
+                LOG.info(
+                        "SmsDeliveryStatus: {}, NotifyStatus: {}, CountryCode: {}",
+                        deliveryStatus,
+                        deliveryReceipt.getStatus(),
+                        countryCode);
+                cloudwatchMetricsService.incrementCounter(
+                        deliveryStatus,
+                        Map.of(
+                                "CountryCode",
+                                String.valueOf(countryCode),
+                                "Environment",
+                                configurationService.getEnvironment(),
+                                "NotifyStatus",
+                                deliveryReceipt.getStatus()));
+                LOG.info("SMS callback request processed");
+            }
+        } catch (JsonProcessingException e) {
+            LOG.error("Unable to parse Notify Delivery Receipt");
+            throw new RuntimeException("Unable to parse Notify Delivery Receipt");
+        }
+        return null;
+    }
+
+    private void validateBearerToken(Map<String, String> headers) {
+        var notifyCallbackBearerToken = configurationService.getNotifyCallbackBearerToken();
+        if (Objects.isNull(headers.get(AUTHORIZATION_HEADER))
+                || !headers.get(AUTHORIZATION_HEADER).startsWith("Bearer ")) {
+            LOG.error("No bearer token in request");
+            throw new RuntimeException("No bearer token in request");
+        }
+        var token = headers.get(AUTHORIZATION_HEADER).substring(7);
+        if (!token.equals(notifyCallbackBearerToken)) {
+            LOG.error("Invalid bearer token in request");
+            throw new RuntimeException("Invalid bearer token in request");
+        }
+    }
+
+    private int getCountryCodeFromNumber(String number) {
+        String defaultRegion = null;
+        if (!number.startsWith("+")) {
```

Comment:
I don't think we need this, I think `defaultRegion` should always be `GB`.

My understanding of how `parse()` works is that if the number starts with `+` then it essentially doesn't use the `defaultRegion` (except when generating the international dialling number, which we're not using anyway).